axon-server-connector/src/main/java/org/axonframework/axonserver/connector/command/AxonServerCommandBusConnector.java

```diff
@@ -0,0 +1,297 @@
+                .putAllMetaData(AxonServerMetadataConverter.convertToGrpcMetaDataValues(command.getMetaData()))
+                .setPayload(SerializedObject.newBuilder()
+                                            .setData(ByteString.copyFrom(payloadAsBytes))
+                                            .setType(command.type().name())
+                                            .setRevision(command.type().version())
+                                            .build())
+                .build();
+    }
+
+    private void addPriority(Command.Builder builder, CommandMessage<?> command) {
+        OptionalInt priority = command.priority();
+        if (priority.isEmpty()) {
+            return;
+        }
+        var instruction = createProcessingInstruction(
+                ProcessingKey.PRIORITY,
+                MetaDataValue.newBuilder().setNumberValue(priority.getAsInt()));
+        builder.addProcessingInstructions(instruction).build();
+    }
+
+    private void addRoutingKey(Command.Builder builder, CommandMessage<?> command) {
+        Optional<String> routingKey = command.routingKey();
+        if (routingKey.isEmpty()) {
+            return;
+        }
+        var instruction = createProcessingInstruction(
+                ProcessingKey.ROUTING_KEY,
+                MetaDataValue.newBuilder().setTextValue(routingKey.get()));
+        builder.addProcessingInstructions(instruction).build();
+    }
+
+    @Override
+    public void subscribe(@Nonnull QualifiedName commandName, int loadFactor) {
+        Assert.isTrue(loadFactor >= 0, () -> "Load factor must be greater than 0.");
+        logger.info("Subscribing to command [{}] with load factor [{}]", commandName, loadFactor);
+        Registration registration = connection.commandChannel()
+                                              .registerCommandHandler(this::incoming, loadFactor, commandName.name());
+
+        // Make sure that when we subscribe and immediately send a command, it can be handled.
+        if (registration instanceof AsyncRegistration asyncRegistration) {
+            try {
+                // Waiting synchronously for the subscription to be acknowledged, this should be improved
+                // TODO https://github.com/AxonFramework/AxonFramework/issues/3544
+                asyncRegistration.awaitAck(2000, TimeUnit.MILLISECONDS);
+            } catch (TimeoutException e) {
+                throw new RuntimeException(
+                        "Timed out waiting for subscription acknowledgment for command: " + commandName, e
+                );
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException("Thread interrupted while waiting for subscription acknowledgment", e);
+            }
+        }
+        this.subscriptions.put(commandName, registration);
+    }
+
+    private CompletableFuture<CommandResponse> incoming(Command command) {
```

Comment:
Perhaps a nit, but shouldn't this method just be called `handle`? From the perspective of the `AxonServerCommandBusConnector`, this is _the_ handle.